packages/service/src/utils/users.ts

```diff
@@ -1,5 +1,48 @@
 import { db, pool, schema } from "../db";
 
-export function createUser(props: { name: string }) {
-  return db.upsert("users", props, "name").run(pool);
+export function createUser(props: { name: string; password: string }) {
+  return db
+    .insert("users", {
+      name: props.name,
+      password: db.sql`crypt(${db.param(props.password)}, gen_salt('md5'))`,
+    })
+    .run(pool);
 }
+
+export function validateUser(props: {
+  name: string;
+  password: string;
+  id: string;
+}) {
+  const user = db.sql<
+    schema.users.SQL,
+    schema.users.Selectable[]
+  >`SELECT (${db.sql`crypt(${db.param(
+    props.password
+  )}, ${"password"})`}) AS ${"password"}, ${"id"}, ${"name"} FROM ${"users"} WHERE ${{
+    id: props.id,
+    name: props.name,
+  }}`.run(pool);
+  return user;
```

Comment:
It would be nice if this function returned a single user instead of a list of users.